hs_composite_resource/models.py

```diff
@@ -254,6 +254,19 @@ def _recreate_xml_docs_for_folder(self, folder, check_metadata_dirty=False):
             for aggregation in aggregations:
                 aggregation.create_aggregation_xml_documents()
 
+    def get_aggregation_by_dataset_name(self, dataset_name):
+        """Get an aggregation that matches the aggregation dataset_name specified by *dataset_name*
+        :param  name: name (aggregation path) of the aggregation to find
+        :return an aggregation object if found
+        :raises ObjectDoesNotExist if no matching aggregation is found
+        """
+        for aggregation in self.logical_files:
+            # remove the last slash in aggregation_name if any
+            if aggregation.dataset_name.rstrip('/') == dataset_name:
```

Comment:
I assume it is possible that multiple aggregations in given resource can have the same dataset name.

Reply:
@pkdash - how are we using the dataset name? Will it appear in URLs in the Resource Map document? I think the answer to your question here is yes, as long as there are not two aggregations in the same folder with the same dataset name because the URL paths to them have to be distinct.

Reply:
He was pointing out a bug in my download code.  I'm updating it to retrieve the aggregation using the pathname.  The URL already includes the pathname in the download request.

Reply:
I was curious how this new function ```get_aggregation_by_dataset_name()``` would work if the dataset name is not unique within a resource.